### PR TITLE
Feature/fetch members by group

### PR DESCRIPTION
**Pull request by StephenMakrogianni**
AC: Only import users in the groups that are specified in the Okta User Group Filter admin.

### CODE REVIEW

File: src/jobs/SyncOktaUsersJob.php [hunk 1]
```diff
@@ -76,16 +77,26 @@ public function getTitle()
...
     public function process()
     {
-        // make an api call to get the list of users from okta
-        $users = Injector::inst()
-            ->get(OktaService::class)
-            ->getAllUsers(100, Config::inst()->get(SyncOktaUsersJob::class, 'statuses_to_sync'));
+        // For each of the user group filters make an api call to get the list of users from okta
+        $allGroupIDs = OktaUserGroupFilter::get();
+        $allUsers = [];
+        foreach ($allGroupIDs as $allGroupID) {
```
Comment (emteknetnz) [inline src/jobs/SyncOktaUsersJob.php]: This is a bit confusing

foreach (OktaUserGroupFilter::get() as $oktaUserGroupFilter) is much more obvious what's going on

File: src/jobs/SyncOktaUsersJob.php [hunk 2]
```diff
@@ -76,16 +77,26 @@ public function getTitle()
...
+            $users = Injector::inst()
+                ->get(OktaService::class)
```
Comment (emteknetnz) [inline src/jobs/SyncOktaUsersJob.php:84]: does this actually work?   presumably equivalent to what it's doing here and reads cleaner would be to put this above the foreach block

$oktaService = Injector::singleton(OktaService::class)

File: src/jobs/SyncOktaUsersJob.php [hunk 3]
```diff
@@ -183,6 +194,10 @@ private function splitUsersIntoCategories(array $users, $uniqueField)
 
         // put users into the category that relates
         foreach ($users as $user) {
+            if (is_array($user)) {
```
Comment (emteknetnz) [inline src/jobs/SyncOktaUsersJob.php:196]: why do we need to check?  should should pretty consistently be the same variable type?

File: src/jobs/SyncOktaUsersJob.php [hunk 4]
```diff
@@ -389,6 +404,9 @@ private function getValueFromUser($user, $key)
         if ($oktaFieldPartsCount == 2) {
             $part0 = $oktaFieldParts[0];
             $part1 = $oktaFieldParts[1];
+            if (is_array($user->$part0)) {
```
Comment (emteknetnz) [inline src/jobs/SyncOktaUsersJob.php:406]: same question as above

File: src/jobs/SyncOktaUsersJob.php (hunk 3 above)
Comment (StephenMakrogianni) [reply src/jobs/SyncOktaUsersJob.php:196]: The modification made returns an array but the rest of the functionality expects an object. Safer to cast as object than modify already working functionality.

File: src/jobs/SyncOktaUsersJob.php (hunk 1 above)
Comment (StephenMakrogianni) [reply src/jobs/SyncOktaUsersJob.php]: Sweet I will modify.

File: src/jobs/SyncOktaUsersJob.php (hunk 4 above)
Comment (StephenMakrogianni) [reply src/jobs/SyncOktaUsersJob.php:406]: Same answer as above

File: src/jobs/SyncOktaUsersJob.php (hunk 2 above)
Comment (StephenMakrogianni) [reply src/jobs/SyncOktaUsersJob.php:84]: Hmmm I gave this a go but it doesn't work.

**Comment by emteknetnz** [comment]
Approved

**Pull request merged**